From: /C=EU/ST=EU/CN=Patrick McHardy
Date: Tue, 15 Jan 2008 15:46:35 +0000 (+0000)
Subject: Fix CONNMARK mask initialisation
X-Git-Tag: iptables_1_4_1_rc1~84
X-Git-Url: https://git.netfilter.org/cgi-bin/gitweb.cgi?p=iptables.old-history.git;a=commitdiff_plain;h=5c6b6c346982c98d2e9797b871556bdcb5ee36ee

Fix CONNMARK mask initialisation

This patch fixes the problem that the CONNMARK mask value
has been set to 0 whenever the CONNMARK target options were
not the last options to be processed.
It initalizes the mask value rather than setting it for
each parse.

Signed-off-by: Peter Warasin <peter@endian.com>
---

diff -uBp a/extensions/libip6t_CONNMARK.c b/extensions/libip6t_CONNMARK.c
--- a/extensions/libip6t_CONNMARK.c
+++ b/extensions/libip6t_CONNMARK.c
@@ -60,6 +60,11 @@ static struct option opts[] = {
 static void
 init(struct ip6t_entry_target *t, unsigned int *nfcache)
 {
+	struct ipt_connmark_target_info *markinfo
+		= (struct ipt_connmark_target_info *)t->data;
+
+	markinfo->mask = 0xffffffffUL;
+
 }
 
 /* Function which parses command options; returns true if it
@@ -72,8 +77,6 @@ parse(int c, char **argv, int invert, un
 	struct ipt_connmark_target_info *markinfo
 		= (struct ipt_connmark_target_info *)(*target)->data;
 
-	markinfo->mask = 0xffffffffUL;
-
 	switch (c) {
 		char *end;
 	case '1':
diff -uBp a/extensions/libipt_CONNMARK.c b/extensions/libipt_CONNMARK.c
--- a/extensions/libipt_CONNMARK.c
+++ b/extensions/libipt_CONNMARK.c
@@ -60,6 +60,10 @@ static struct option opts[] = {
 static void
 init(struct ipt_entry_target *t, unsigned int *nfcache)
 {
+	struct ipt_connmark_target_info *markinfo
+		= (struct ipt_connmark_target_info *)t->data;
+
+	markinfo->mask = 0xffffffffUL;
 }
 
 /* Function which parses command options; returns true if it
@@ -72,8 +76,6 @@ parse(int c, char **argv, int invert, un
 	struct ipt_connmark_target_info *markinfo
 		= (struct ipt_connmark_target_info *)(*target)->data;
 
-	markinfo->mask = 0xffffffffUL;
-
 	switch (c) {
 		char *end;
 	case '1':
